app/search/page.tsx

```diff
@@ -0,0 +1,60 @@
+import { MATCH_TYPE } from 'constant';
+import { notFound, redirect } from 'next/navigation';
+import { getSearchResult, getWalletSwaps } from 'services';
+import Result from './_components/Result';
+
+export async function generateMetadata({
+  searchParams,
+}: {
+  searchParams: { query: string };
+}) {
+  return {
+    title: `Address ${searchParams.query}`,
+  };
+}
+
+const Page = async ({
+  searchParams,
+}: {
+  searchParams: { query?: string; page?: string };
+}) => {
+  const { query, page } = searchParams;
+
+  if (!query) notFound();
+
+  let result;
+  if (typeof query === 'string') {
+    result = await getSearchResult(query);
+  }
+
+  if (!result?.length) notFound();
+
+  if (result[0].matchType === MATCH_TYPE.REQUESTID) {
+    redirect(`/swap/${query}`);
+  }
+
+  if (result[0].matchType === MATCH_TYPE.TXHASH) {
+    const { requestId } = result[0];
+    if (requestId) redirect(`/swap/${requestId}`);
+  }
+
+  const data = await getWalletSwaps(query, page);
+  const { transactions, total } = data || {};
+
+  if (!data || !transactions?.length) {
+    notFound();
+  }
+
+  return (
+    <div className="w-full flex justify-center">
+      <Result
```

Comment:
Please use `Suspense` for streaming data like the home page.

Reply:
Done in https://github.com/rango-exchange/explorer/pull/32/commits/914e71cc3f271a82b030c10c29d3e95418395dbe